app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java

```diff
@@ -107,14 +107,24 @@ protected void setBarVisible(ViewGroup barLayout, boolean visible) {
         }
     }
 
-    protected int findLineStart(int cursor, String text) {
+    protected int findLineStart(int cursor, String text, Boolean ignoreLeadingWhitespace) {
         int i = cursor - 1;
+
         for (; i >= 0; i--) {
             if (text.charAt(i) == '\n') {
                 break;
             }
         }
 
+        if (ignoreLeadingWhitespace) {
+            for (; i < (cursor - 1); ++i) {
+                char c = text.charAt(i + 1);
+                if (c != ' ' && c != '\t') {
+                    break;
+                }
+            }
+        }
```

Comment:
Linear search forward from newline. Should not cause noticeable performance hit.